### PR TITLE
feat: support more sensitive fields for encryption

### PR DESCRIPTION
**Pull request by shreemaan-abhishek**
### Description

Some sensitive plugin fields aren't under the protection of `encrypt_fields` feature. It has been addressed in this PR.

### Checklist

- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: t/plugin/authz-casdoor.t [hunk 1]
```diff
@@ -506,6 +506,8 @@ apisix:
             -- get plugin conf from etcd, password is encrypted
             local etcd = require("apisix.core.etcd")
             local res = assert(etcd.get('/routes/1'))
+            local core = require("apisix.core")
+            core.log.warn("dibag", core.json.encode(res.body.node.value.plugins["authz-casdoor"].client_secret))
```
Comment (starsz) [inline t/plugin/authz-casdoor.t:510]: Debug line ?

Comment (shreemaan-abhishek) [reply t/plugin/authz-casdoor.t:510]: oops, let me remove it.

File: apisix/plugins/jwe-decrypt.lua [hunk 2]
```diff
@@ -71,15 +72,26 @@ function _M.check_schema(conf, schema_type)
...
-        else
-            if #conf.secret ~= 32 then
-                return false, "the secret length should be 32 chars"
+        local local_conf, err = core.config.local_conf(true)
+        if not local_conf then
+            return false, "failed to load the configuration file: " .. err
+        end
+
+        local encrypted = core.table.try_read_attr(local_conf, "apisix", "data_encryption",
+        "enable_encrypt_fields") and (core.config.type == "etcd")
```
Comment (starsz) [inline apisix/plugins/jwe-decrypt.lua:81]: Why do we need to add this ? 
I think the `check_schema` is run before the fields encrypted.

Comment (shreemaan-abhishek) [reply apisix/plugins/jwe-decrypt.lua:81]: if data encryption is enabled then the secret length will be more than 32. So we should not check the length if data encryption is on.

> I think the check_schema is run before the fields encrypted.

yes. This is why we cannot use the code in `plugin.lua`, so I just copied the logic 😅

**Review by moonming — APPROVED** [review]

File: apisix/plugins/jwe-decrypt.lua (hunk 2 above)
Comment (starsz) [reply apisix/plugins/jwe-decrypt.lua:81]: > if data encryption is enabled then the secret length will be more than 32.

Why the secret length will be more than 32. Is the secret is encrypted?

Comment (shreemaan-abhishek) [reply apisix/plugins/jwe-decrypt.lua:81]: yes, exactly.

**Review by starsz — APPROVED** [review]

**Pull request merged**